cadence/contracts/bridge/FlowEVMBridge.cdc

```diff
@@ -464,21 +765,36 @@ contract FlowEVMBridge : IFlowEVMNFTBridge {
     access(self)
     fun deployDefiningContract(evmContractAddress: EVM.EVMAddress) {
         // Deploy the Cadence contract defining the asset
-        // Treat as NFT if ERC721, otherwise FT
+        // Treat as NFT if contract is ERC721, otherwise treat as FT
         let name: String = FlowEVMBridgeUtils.getName(evmContractAddress: evmContractAddress)
         let symbol: String = FlowEVMBridgeUtils.getSymbol(evmContractAddress: evmContractAddress)
+        let contractURI = FlowEVMBridgeUtils.getContractURI(evmContractAddress: evmContractAddress)
+        var decimals: UInt8 = FlowEVMBridgeConfig.defaultDecimals
 
         // Derive contract name
-        let isERC721: Bool = FlowEVMBridgeUtils.isEVMNFT(evmContractAddress: evmContractAddress)
-        let cadenceContractName: String = FlowEVMBridgeUtils.deriveBridgedNFTContractName(from: evmContractAddress)
-        let contractURI = FlowEVMBridgeUtils.getContractURI(evmContractAddress: evmContractAddress)
+        let isERC721: Bool = FlowEVMBridgeUtils.isERC721(evmContractAddress: evmContractAddress)
+        var cadenceContractName: String = ""
+        if isERC721 {
+            // Assert the contract is not mixed asset
+            let isERC20 = FlowEVMBridgeUtils.isERC20(evmContractAddress: evmContractAddress)
+            assert(!isERC20, message: "Contract is mixed asset and is not currently supported by the bridge")
+            // Derive the contract name from the ERC721 contract
+            cadenceContractName = FlowEVMBridgeUtils.deriveBridgedNFTContractName(from: evmContractAddress)
+        } else {
```

Comment:
Might want an explicit check and panic here too. It stops any errors to fall through the cracks.

Reply:
Do you mean check `isERC20`? Further up the call stack, `isERC20` would've been called
```
onboardByEVMAddress
|_evmAddressRequiresOnboarding
  |_FlowEVMBridgeUtils.isValidEVMAsset
    |_isERC20
...
|_deployDefiningContract
```
When I was testing locally, I found the EVM redundant calls consumed a lot of gas and made a pass at minimizing them.. But if we feel the explicit checks are worth it I can add them back in.